elasticdl_preprocessing/layers/to_ragged.py

```diff
@@ -0,0 +1,72 @@
+import tensorflow as tf
+
+_COMMA_SEP = ","
+
+
+class ToRagged(tf.keras.layers.Layer):
+    """Converts a `Tensor` or `RaggedTensor` to a `RaggedTensor`,
+    dropping ignore_value cells. If the input's dtype is string, split
+    the string elements to convert the input to `RaggedTensor` firstly.
+
+    Note that the TensorFlow version with the layer must be greater than 2.0.0.
+
+    Example (Integer):
+    ```python
+        layer = ToRagged()
+        inp = tf.constant([[1], [-1], [4]], tf.int64)
+        out = layer(inp)
+        [[1], [], [4]]
+    ```
+
+    Example (String):
+    ```python
+        ayer = ToRagged()
+        inp = tf.constant([["1,2,3"], ["4,5"], [""]])
+        out = layer(inp)
+    ```
+    The expected output is `[["1", "2", "3"], ["4", "5"], []]`
+
+    Arguments:
+        sep: Valid if the input's dtype is string.
+        ignore_value: Entries in inputs equal to this value will be
+            absent from the output `RaggedTensor`. If `None`, default value of
+            input's dtype will be used ('' for `str`, -1 for `int`).
+
+    Input shape: A numeric or string `Tensor` or `RaggedTensor` of shape
+        `[batch_size, d1, ..., dm]`
+
+    Output shape: An `RaggedTensor` with the same shape as inputs
+    """
+
+    def __init__(self, sep=_COMMA_SEP, ignore_value=None):
+        super(ToRagged, self).__init__()
+        self.sep = sep
+        self.ignore_value = ignore_value
+
+    def call(self, inputs):
+        if not isinstance(inputs, tf.Tensor) and not isinstance(
+            inputs, tf.RaggedTensor
+        ):
+            raise TypeError(
+                "The inputs must be a Tensor or RaggedTensor and "
+                "the type of inputs is {}".format(type(inputs))
+            )
+
+        if isinstance(inputs, tf.Tensor):
+            inputs = tf.RaggedTensor.from_tensor(inputs)
+
+        ignore_value = self.ignore_value
+        if ignore_value is None:
+            if inputs.dtype == tf.string:
+                ignore_value = ""
+                inputs = tf.strings.split(inputs, sep=self.sep).values
+            elif inputs.dtype.is_integer:
+                ignore_value = -1
```

Comment:
what about other `dtype`? It seems that it is not complete. And you can add a process function for `ignore_value` in order to test and it everywhere. All you have to do is to gather enough information and return a valid `ignore_value`.

Reply:
Yeah, Done!